lib/dockerregistry/checks.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strings"
+
+	gogit "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// ImageRemovalCheck implements the PreCheck interface and checks against
+// pull requests that attempt to remove any images from the promoter manifests.
+type ImageRemovalCheck struct {
+	GitRepoPath    string
+	MasterSHA      plumbing.Hash
+	PullRequestSHA plumbing.Hash
+}
+
+func isValidSha(potenitalSHA string) error {
+	sha256Length := 64
+	if len(potenitalSHA) == sha256Length {
```

Comment:
Typically, the idiom I've seen in imperative languages (Python, Java, also Go) is to avoid nesting if-conditions too much. This is the style you've already used in the Run() function with its various Git ops and error-checking for each result.

E.g.
```
if a {
  return err
}

if b {
  return err
}

return nil
```

Can you apply it here?